### PR TITLE
Extra encounters and forging for all protagonists

### PR DESCRIPTION
**Pull request by bulian-bg**
well that take me about 3 hours, good thing nap time went long!

### CODE REVIEW

File: ImprovedAnvil/lib/optional/all_protagonists_extra_rewards.TPH [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Remove class restrictions from nonessential kiit drios
+
+// Golden Girdle in Watcher's Keep level 4
+COPY_EXISTING ~override\AR3017.BCS~ ~override\AR3017.BCS~
+  DECOMPILE_AND_PATCH BEGIN
+    REPLACE_TEXTUALLY ~!Class(Player1,MAGE_ALL)~ ~~
+  END
+
+// Scroll of Memory Boosting in Spellhold
```
Comment (critto-bg) [inline ImprovedAnvil/lib/optional/all_protagonists_extra_rewards.TPH:9]: these drops are protagonist-based treasure: vagrant gets cloak of magic shielding, mages and druids - a scroll of memory boosting, the rest get a barbarian essence potion.
in its current form this block messes stuff up. I'm dropping it for a moment. Let me know what you want to do here, and I'll reimplement it properly.

File: ImprovedAnvil/lib/optional/all_protagonists_extra_rewards.TPH [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+COPY_EXISTING ~override\AR154.BCS~ ~override\AR1514.BCS~
+  DECOMPILE_AND_PATCH BEGIN
+    REPLACE_TEXTUALLY ~IF Global("Iaitem","AR1514",0) Global("closedDoor","AR1514",2) OR(2) Class(Player1,MAGE_ALL) Class(Player1,DRUID_ALL) !Class(Player1,FIGHTER_ALL) !Class(Player1,THIEF_ALL) !Class(Player1,CLERIC_ALL) THEN RESPONSE #100 SetGlobal("Iaitem","AR1514",1) GiveItemCreate("S!misc01",Player1,0,0,0) END~ ~IF Global("Iaitem1","AR1514",0) Global("closedDoor","AR1514",2) THEN RESPONSE #100 SetGlobal("Iaitem1","AR1514",1) GiveItemCreate("S!misc01",Player1,0,0,0) END~ 
+  END
+
+// Scrolls in Planar Sphere
+COPY_EXISTING ~override\AR0410.BCS~ ~override\AR0410.BCS~
```
Comment (critto-bg) [inline ImprovedAnvil/lib/optional/all_protagonists_extra_rewards.TPH:16]: this is actually more complicated than you think
I've made a patch a year ago that made the scrolls add up continuously without the need for the player to come into the sphere after each batch in order to reset the counter. I've had to add patching of BALDUR.BCS as well.

**Comment by critto-bg** [comment]
I'm 3/4 through your commit. Some bad news: pretty much none of it is written correctly, except for the simplest one-liners. When you do a REPLACE_TEXTUALLY, it does a verbatim search unless you use a regexp. Multi-liners aren't matched in none of your blocks. I'm having to rewrite this with regular expressions. Thankfully, it's pretty easy, but I still have to re-check each block to make sure.

Some things are just weird or missing crucial steps, I mention those in separate comments.

While I'm at it, I want to ask if you checked the dialogues with respective creatures involved in all of those quests you've made available for all protags. It's quite possible that some of the dialogues involve kit checks as well. I didn't look into it personally. I see that you've made replacements in crom/cespy dialogue, so I assume you did some searches through dialogues as well, but I ask anyway to be sure.

File: ImprovedAnvil/lib/optional/all_protagonists_extra_rewards.TPH (hunk 1 above)
Comment (critto-bg) [reply ImprovedAnvil/lib/optional/all_protagonists_extra_rewards.TPH:9]: Nevermind, I know what to do here

Comment (bulian-bg) [reply ImprovedAnvil/lib/optional/all_protagonists_extra_rewards.TPH:9]: i'm not sure what you did but i changed the variable name for the scroll in the non-essential loot addon, so in principle the party would get both the cloak and scroll, and the barbarian essence potion drop essentially never happens.  that seemed to make sense to me - what did you end up doing?

**Comment by bulian-bg** [comment]
yeah, not too surprising.  i was pretty sure the syntax was incorrect.  unfortunately, the scripts were complicated enough, particularly the planar sphere one, that simple deletions weren't possible.  this is probably a case where just having a separate file for install would have been significantly simpler in the end

there are no class checks in the dialogues, fortunately.

File: ImprovedAnvil/lib/optional/all_protagonists_extra_rewards.TPH (hunk 2 above)
Comment (bulian-bg) [reply ImprovedAnvil/lib/optional/all_protagonists_extra_rewards.TPH:16]: good point, i didn't patch baldur.bcs and it looks like there is a ranger quest (684) line as well as the mage scroll patch there (1007).

**Comment by critto-bg** [comment]
I'm closing this one since you've pushed the patch into the appropriate repo.

**Pull request closed without merging**